internal/tiltfile/encoding/yaml.go

```diff
@@ -35,11 +38,26 @@ func readYAML(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple
 		return nil, err
 	}
 
-	return yamlStringToStarlark(string(contents), path.GoString())
+	return yamlStreamToStarlark(string(contents), path.GoString())
+}
+
+func readYAML(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	v, err := readYAMLStream(thread, fn, args, kwargs)
+	if err != nil {
+		return nil, err
+	}
+	l, ok := v.(*starlark.List)
```

Comment:
have you considered an intermediate function that returns the concrete type, so that you don't need to do the type-casting square dance?

Reply:
nope, that's a good idea